src/map/lua/lua_baseentity.cpp

```diff
@@ -7268,6 +7268,24 @@ void CLuaBaseEntity::delLearnedWeaponskill(uint8 wsID)
     PChar->pushPacket(new CCharAbilitiesPacket(PChar));
 }
 
+/************************************************************************
+ *  Function: trySkillUp()
+ *  Purpose : Attempts to increase skill for <skill> based on compared mob <level>
+ *  Example : player:trySkillUp(tpz.skill.HAND_TO_HAND, 50)
+ *  Notes   :
+ ************************************************************************/
+
+void CLuaBaseEntity::trySkillUp(uint8 skill, uint8 level)
+{
+    if (m_PBaseEntity->objtype == TYPE_PC)
+    {
+        auto* PChar = static_cast<CCharEntity*>(m_PBaseEntity);
+        return charutils::TrySkillUP(PChar, static_cast<SKILLTYPE>(skill), level);
```

Comment:
The function is marked as void, but you're returning, so it may or may not do anything

Reply:
Ah, it's meant to be void, this just needs to be `charutils::TrySkillUP(PChar, static_cast<SKILLTYPE>(skill), level);`